constellations/gemini-admin/src/achieve/achieveEdit.js

```diff
@@ -143,35 +115,66 @@ export class AchieveEditPage extends React.Component {
                 />
             );
         }
+        return modalDiv;
+    };
+
+    render = () => {
+        const isEdit = this.state.isEdit;
+        const title = isEdit ? "Edit Achievement" : "Add Achievement";
+
+        let deleteButton = isEdit ? (
+            <button className="btn-delete" onClick={this.onClickDelete}>
+                Delete
+            </button>
+        ) : (
+            <div></div>
+        );
 
         return (
             <div id="view-achieve-edit">
-                {modalDiv}
+                {this.renderModal()}
                 <h2>{title}</h2>
+
                 <InputText
                     label="Year"
                     description="Enter the achievement year"
                     required={true}
-                    value={this.state.inputYear}
-                    onChangeCallback={(e) => this.handleChange(e, "inputYear")}
+                    value={this.state.year}
+                    onChangeCallback={(e) => this.handleChange(e, "year")}
                     validators={[
                         {
-                            validate: (text) => parseInt(text) > 2000,
+                            validate: (number) => parseInt(number) > 2000,
                             message: "You must input a year greater than 2000",
                         },
                     ]}
                 />
+
+                <InputText
+                    label="Position"
+                    description="Enter the position (Lower position numbers are shown first in that year)"
+                    required={true}
+                    value={this.state.position}
+                    onChangeCallback={(e) => this.handleChange(e, "position")}
+                    validators={[
+                        {
+                            validate: (number) =>
+                                Number.isInteger(parseInt(number)) &&
+                                parseInt(number) > 0,
```

Comment:
make position always be a number, default 0 up in the `state` variable definition.

Reply:
If position is NOT a number, we WANT to get errors

Reply:
If it is not a number, it does throw an error right now.

Reply:
This is me forgetting that things being inputted come as a string first. You can disregard my comment. My bad, sorry.